pkg/cloud/scope/cluster.go

```diff
@@ -98,6 +100,23 @@ type ClusterScope struct {
 	controllerName  string
 }
 
+func (s *ClusterScope) KubeadmControlPlaneBindPort() int32 {
+	if s.Cluster.Spec.ControlPlaneRef != nil {
+		cpObjectKey := client.ObjectKey{
+			Namespace: s.Cluster.Namespace,
+			Name:      s.Cluster.Spec.ControlPlaneRef.Name,
+		}
+		var kubeadmCP controlplanev1.KubeadmControlPlane
+		_ = s.client.Get(context.TODO(), cpObjectKey, &kubeadmCP)
+
+		var apiEndpoint bootstrapv1.APIEndpoint
+		if kubeadmCP.Spec.KubeadmConfigSpec.InitConfiguration != nil && kubeadmCP.Spec.KubeadmConfigSpec.InitConfiguration.LocalAPIEndpoint != apiEndpoint {
```

Comment:
```suggestion
		if kubeadmCP.Spec.KubeadmConfigSpec.InitConfiguration != nil && kubeadmCP.Spec.KubeadmConfigSpec.InitConfiguration.LocalAPIEndpoint != bootstrapv1.APIEndpoint{} {
```

Reply:
Adding this gives the compile time error, Type 'bootstrapv1.APIEndpoint' is not an expression. That's why added the variable and then compared.

Reply:
is there any better way to do this?

Reply:
Can use `reflect.DeepEqual(a,b)`

Reply:
Since this is just a nil check and not any complex structure where we have to compare each and every attribute of the struct with each other, i think it should be fine, wdyt?

Reply:
Yes your point is right as its just a nil check so we can ignore it but in case if you want to avoid creating new variable then `reflect.DeepEqual(a,b)` works.